src/main/scala/mesosphere/marathon/core/storage/zookeeper/PersistenceStore.scala

```diff
@@ -53,18 +53,20 @@ trait PersistenceStore {
     *
     * @return
     */
-  def childrenFlow: Flow[String, Seq[String], NotUsed]
-  def children(path: String): Future[Seq[String]]
-  def children(paths: Seq[String]): Source[Seq[String], NotUsed] = Source(paths).via(childrenFlow)
+  type Children = Seq[String]
```

Comment:
Should be type the Children name? Such as `PathComponent`, `Segment` or `Path`?

Also, let's say I have

```
/aws
  /prod
  /dev
    /ci
      /123
```

What is the content of `children("/aws/dev")`? Is it `Seq("/aws/dev/ci", "/aws/dev/ci/123")` or `Seq("ci", "123")`?

Reply:
> Should be type the Children name? Such as `PathComponent`, `Segment` or `Path`?

I don't think I understand the question. Mostly I wanted to avoid `Source[Try[Seq[String]` in the method signature

> What is the content of `children("/aws/dev")`?

it's always the direct children of the node (that's how ZK returns it) so only `/ci` and if you want the absolute path (`absolute = true`) it is `/aws/dev/ci`

Reply:
> Should be type the Children name? Such as `PathComponent`, `Segment` or `Path`

I meant that we should type them. So `Seq[Segment]` instead of `Seq[String]`.

Reply:
Noo, then e.g. `create` should also take a change in the signature. In general this layer deals with strings (for keys) and bytestrings (for values). I think it's fine like this

Reply:
I guarantee that this will lead to lots of bugs. I don't see why `Path` and `Path.segments: Seq[Segment]` is such a bad idea. I mean the whole thing is design around paths. `String` is nice and simple at the beginning. Well, We probably gonna use `PathId` anyways to templates...

Reply:
No, this is definitely a wrong layer for introducing _PathIds_ or _Path.Segments_. This is the layer, closest to ZK, everything here is "native" e.g. _strings_ and _byte[]_. Pulling in higher level concerns in this layer would be a leaking abstraction:
```
 +------------------------------------+
 |            Zookeeper               |
 +------------------------------------+
                  |
                  |
 +----------------v-------------------+
 |         PersistenceStore           |
 |                                    |
 |  - uses apache curator             |
 |  - deals with ZK nodes,            |
 |    strings and byte[] payloads     |
 +------------------------------------+
                  |
                  |
                  |
 +------------------------------------+
 |          Repository                |
 |                                    |
 |   - uses PersistenceStore          |
 |   - deals with application level   |
 |     objects e.g. Templates         |
 |   - encode/decode to/from ZK nodes |
 |                                    |
 +------------------------------------+

```